docs/core-components.md

```diff
@@ -0,0 +1,93 @@
+* `oef`: this protocol provides the AEA protocol implementation for communication with the OEF including an `OEFMessage` class for hooking up to OEF services and search agents. Utility classes are available in the `models.py` module which provides OEF specific requirements such as classes needed to perform querying on the OEF such as `ConstraintExpr`, `Query`, `And`, and `Or`, to name a few.
+* `fipa`: this protocol provides classes and functions needed for AEA agent communication via the FIPA Agent Communication Language. For example, the `FIPAMessage` class provides negotiation terms such as `cfp`, `propose`, `decline`, etc. Protobuf configuration is also available.
+
+### Connection
+
+Connections wrap an external SDK or API and manage the messaging. They allow the agent to connect to an external service which has a Python SDK or API. 
+
+The module `connections/base.py` contains two abstract classes which define a `Channel` and a `Connection`. A `Connection` contains one `Channel`.
+
+The framework provides a number of default connections.
+
+* `local`: implements a local node.
+* `oef`: wraps the OEF SDK.
+
+### Skill
+
+Skills deliver economic value to the AEA by allowing agents to encapsulate and call any kind of code. It encapsulates the abstract base classes `Handler`, `Behaviour`, and `Task`.
+
+* `Handler`: each skill has a single `Handler` which is responsible for the registered protocol messaging. Handlers implement reactive behaviour; by understanding the requirements contained in Envelopes, the Handler reacts appropriately to message requests. 
+* `Behaviour`: one or more Behaviours encapsulate sequences of actions that cause interactions with other agents initiated by the framework. Behaviours implement proactive behaviour.
+* `Task`: one or more Tasks encapsulate background work internal to the agent.
+
+
+
+## Agent 
+
+### Main loop
+
+The `_run_main_loop()` function in the `Agent` class performs a series of activities while the `Agent` state is not stopped.
+
+* `act()`: this function calls the `act()` function of all registered Behaviours.
+* `react()`: this function grabs all Envelopes waiting in the `InBox` queue and calls the `handle()` function on them.
+* `update()`: this function loops through all the Tasks and executes them.
+
+
+## Decision maker
+
+The DecisionMaker component manages global agent state updates proposed by the skills and processes the resulting ledger transactions.
+
+It is responsible for crypto-economic security.
+
+
+## Filter
+
+The `Filter` routes messages to the correct `Handler` via the `Resource` component.
+
+
+## Resource 
+
+The `Resource` component is made up of Registries which contain Resources. There is one Registry for each type of Resource. 
+
+Message Envelopes travel through the `Filter` which directs them to the correct `Registry`.
```

Comment:
change to: which fetches the correct `Handler` from the `Registry`.

Reply:
Done.